tldw_Server_API/app/core/MCP_unified/protocol.py

```
@@ -1441,6 +1480,27 @@ def _mask_secrets(self, text: str) -> str:
+        original = str(exc)
+        masked = self._mask_secrets(original)
+        if masked == original:
+            with contextlib.suppress(Exception):
+                setattr(exc, "_mcp_masked_secret", False)
+            return exc
+        try:
+            sanitized = exc.__class__(masked)
+        except Exception:
+            sanitized = RuntimeError(masked)
+        with contextlib.suppress(Exception):
+            setattr(sanitized, "_mcp_masked_secret", True)
```

Ruff (0.15.9)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)